File: go/pkg/calculator/calculator_test.go
package calculator

import (
	"testing"
)

func TestNoteName(t *testing.T) {
	tests := []struct {
		x, y     int
		expected string
	}{
		{49, 55, "Cb77v+"},
		{31, 16, "Sorry - that is a 31-limit pitch!"},
		{3, 2, "G"},
		{5, 4, "E"},
		{7, 4, "Bb7"},
		{11, 8, "F^"},
		{13, 8, "Ab3"},
	}

	for _, tt := range tests {
		result := NoteName(tt.x, tt.y, true)
		if result != tt.expected {
			t.Errorf("NoteName(%d, %d) = %q, expected %q", tt.x, tt.y, result, tt.expected)
		}
	}
}

File: ts/tests/calculator.test.ts
import { noteName } from '../src/calculator';

describe('noteName', () => {
    test('examples from README', () => {
        expect(noteName(49, 55, true)).toBe("Cb77v+");
        expect(noteName(31, 16, true)).toBe("Sorry - that is a 31-limit pitch!");
    });

    test('common ratios', () => {
        expect(noteName(3, 2, true)).toBe("G");
        expect(noteName(5, 4, true)).toBe("E");
        expect(noteName(7, 4, true)).toBe("Bb7");
        expect(noteName(11, 8, true)).toBe("F^");
        expect(noteName(13, 8, true)).toBe("Ab3");
    });
});
